cerulyean ip

cerulyean opened the pull request:
# CirnoBot

>Yukkuri shiteitte ne! - [Yukkuri](https://myanimeshelf.com/figures/2716027_Touhou_Project_Plush_Yukkuri_Reimu_%28Deka_Size%29)

Ever wanted a Yukkuri to help you keep track of your tasks? Doesn't matter, here it is anyway!

- Keeps tracks of tasks!
- Yells at you when you enter commands wrongly!
- Speaks in a vaguely Yukkuri-like manner!
- Catch a glimpse of the rare fumo!

All you need to do is,

1. Download it!
2. Double-click it!
3. Add your tasks!
4. Let it manage your tasks for you! 😉

And it is FREE!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] ~~Reminders~~

If you are a Java programmer, you can use it to practice `Java` too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

> [!TIP]
> Enter help into the text box and click enter to get a list of commands!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

rfcheong9292 committed:
Level-0

rfcheong9292 committed:
Added the echo stuff

rfcheong9292 committed:
added the helper listmaster class with its methods

rfcheong9292 committed:
added the marking for the list

rfcheong9292 committed:
added the todo, deadline and event

rfcheong9292 committed:
added some error message and automtaed testing thing but the automated testing doesnt even work properly will probably mess with it next time

rfcheong9292 committed:
added the ability to delete tasks

rfcheong9292 committed:
added saving

rfcheong9292 committed:
added localtime stuff

rfcheong9292 committed:
small test

rfcheong9292 committed:
small test2

rfcheong9292 committed:
revert

cerulyeann committed:
refactored, more classes

cerulyeann committed:
erge branch 'branch-A-MoreOOP'
Merging stuff lol

cerulyeann committed:
packages added

cerulyeann committed:
Merge branch 'branch-A-Packages'

cerulyeann committed:
Merge branch 'add-gradle-support'

cerulyeann committed:
some tests

cerulyeann committed:
Merge branch 'branch-A-JUnit'

cerulyeann committed:
added some documentation on methods and classes in Doot.java. Also moved the logo to its own file so it doesnt block the entire screen everytime i open the Doot

cerulyeann committed:
Merge branch 'branch-A-JavaDoc'

cerulyeann committed:
added the find command

cerulyeann committed:
Merge branch 'branch-Level-9'

jon-seana commented on src/main/java/doot/Doot.java:
```diff
@@ -0,0 +1,591 @@
+            }
+        }
+
+        //called by the parser when the user enters something starting with mark. It sets the task in TaskList to be marked
```
Good job for including the JavaDoc, but this is the wrong format.
You may follow this format:
/**
* <description>

* @param <description>
* @return <description>
*/

jon-seana commented on src/main/java/doot/Doot.java:
```diff
@@ -0,0 +1,591 @@
+        }
+    }
+
+    public enum Type {T, D, E}
```
Should this deserve a comment to explain what it does?

jon-seana commented on src/main/java/doot/Doot.java:
```diff
@@ -0,0 +1,591 @@
+
+    //the class for storing information on individual tasks.
+    //contains state, description, type. This is an abstract class so that the more specific task classes can inherit
+    public abstract static class Task {
```
Good usage of abstract class!! But perhaps its best to put this Task class into a separate file.

jon-seana commented on src/main/java/doot/Doot.java:
```diff
@@ -0,0 +1,591 @@
+            return list.toString();
+        }
+    }
+    public static class EventTask extends Task {
```
I like how you extend each of the types of tasks (todo, deadline, events) to extend from task, demonstrating good OOP principle. But perhaps you may have missed a JavaDoc comment to explain what EventTask class does.

jon-seana commented on src/main/java/doot/Doot.java:
```diff
@@ -0,0 +1,591 @@
+package doot;
```
Overall, its a neat class code base, but you have missed putting the correct JavaDoc format in each of the public methods and classes.

Also, I highly recommend for you to extract each classes into another file and relevant packages, as to align with the aim of moreOOP increment level in week 3 task.

jon-seana reviewed (commented):
Looks good to me, interms of neatness, although some occasional mis-spacing. 
Some OOP and JavaDoc changes requested too.

LuBolin commented on src/main/java/doot/Doot.java:
```diff
@@ -0,0 +1,591 @@
+
+
+public class Doot {
+    static String[] formats = {
```
Would making this array final make sense?

LuBolin commented on src/main/java/doot/Doot.java:
```diff
@@ -0,0 +1,591 @@
+
+    //the main structure that contains the tasks. Contains methods to modify the tasks' statuses
+    public static class TaskList {
+        private List<Task> arr;
```
As this is a collection of objects (Tasks), plural would be better.
I understand that you cannot simply add s behind arr, so we can maybe consider alternatives such as ArrayOfTasks?

LuBolin commented on src/main/java/doot/Doot.java:
```diff
@@ -0,0 +1,591 @@
+        }
+
+        //returns the entire list, in the format used for printing as a list when called by the user
+        public String returnList() {
```
Despite the JavaDoc, the function name may be misleading for what this function is doing.
Specifically, this may be interpreted as a getter for the list, while in reality it returns a formatted string.
Maybe getFormattedList would be better?

LuBolin commented on src/main/java/doot/Doot.java:
```diff
@@ -0,0 +1,591 @@
+
+        protected LocalDate dateend;
+
+        public EventTask(String description, String start, String end) {
```
Good job using parameter names that have meaning!

LuBolin reviewed (commented):
In terms of "other code quality guidelines", this is great!
I am intrigued by your decision to put all classes in one script.

cerulyean committed:
Update README.md

cerulyeann committed:
actually split my classes across folders

cerulyeann committed:
Merge remote-tracking branch 'origin/master'

cerulyeann committed:
made some of the code follow the style fixes but wow theres alot of problems

cerulyeann committed:
forgot to branch so im trying to fix it

cerulyeann committed:
actually fixed some minor stuff in the DootTest.java thing

cerulyeann committed:
fixing gigignore stuff

cerulyeann committed:
halfway through getting everything to return a string instead of printing everything into terminal

cerulyeann committed:
got the gui working, i will probably change the name and personality of the thing

cerulyeann committed:
.gitignore update

cerulyeann committed:
added a few assertions but i want to go do refactoring instead

cerulyeann committed:
made a Command interface

cerulyeann committed:
cleaned up codequality, broke down some giant methods

cerulyeann committed:
cleaned up codequality, broke down some giant methods

cerulyeann committed:
stop tracking list.txt

cerulyeann committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	data/list.txt
#	src/main/java/doot/Parser.java

cerulyeann committed:
cleaned up codequality, broke down some giant methods

cerulyeann committed:
made the gui nicer and made the thing cirno

cerulyeann committed:
small modification to the test so it doesnt break

cerulyeann committed:
Added the C-Tagging feature by using /tag at the end of the command

cerulyeann committed:
Added the C-Tagging feature by using /tag at the end of the command

cerulyeann committed:
Merge tag 'BCD-Extension'

cerulyeann committed:
Added several more assertions in some classes

cerulyeann committed:
Added several more assertions in some classes

cerulyeann committed:
Added some more javadocs

cerulyeann committed:
Task's makeTask(): added more error handling

Now the task.makeTask() method checks if there are duplicate keywords where there shouldn't be, like /tag or /end.

Parser's keywordCheck(), newly added for the error checking

This method was added to make the error checking cleaner

cerulyeann committed:
Merge branch 'refs/heads/A-Assertions'

cerulyeann committed:
DootTest class: testMakeTask()

I added a testcase which tests the basics of creating 3 simple tasks of each type, and asserting their details are the same.

Task.makeTask(): fixed keyword checking

I changed the keywords checks from /start and /end to /to and /from.

cerulyeann committed:
fix(test): remove redundant InvalidFormatException in testMakeTask()

The InvalidFormatException thrown by testMakeTask() was removed as it is now caught within the function.

cerulyeann committed:
test(DootTest): add testCreationString() to verify task creation string

Added a new test, testCreationString(), to ensure the creation string is correct for basic tasks.

cerulyeann committed:
refactor(Parser): reformat isMark() for improved readability

Refactored the isMark() method to make the code less "arrow-shaped" and easier to read.

cerulyean committed:
Update README.md

Added a more descriptive PR description.

cerulyeann committed:
fix(Storage): improve file corruption error handling

The bot now displays an error message when a file is corrupted, warning the user to check for important data. Previously, the bot would override corrupted data.

fix(HandleDeleteCommand): improve error handling

fix(HandleDeleteCommand): improve error handling for non-integer inputs

The bot now displays an error message when a delete command is entered with non-integer values. Previously, the bot would ignore the command.

cerulyeann committed:
fix(Storage): fix data loading issue

The previous commit broke data loading because it didn't update the parser's list attribute, causing the bot to assume the file was empty.

cerulyeann committed:
fix(Storage): fix data writing issue

The previous commit broke data writing because it didn't update the savelist's file location.

Added image Ui.png.

cerulyeann committed:
Added UserGuide

Included a short guide on how to use the bot, which includes most of it's commands.

cerulyeann committed:
Added UserGuide

Included a short guide on how to use the bot, which includes most of it's commands.

cerulyeann committed:
fix (MainWindow): Fixed bye not working

Previously bye did not close the program, now it will.

cerulyeann committed:
Removed some errors.

cerulyeann committed:
Merge remote-tracking branch 'origin/master'

cerulyeann committed:
Removed redundant data file

cerulyeann committed:
Removed redundant data file

cerulyeann committed:
MainWindow.fxml: Added a background Image

Added a background Image into MainWindow.fxml. That image is sotred in resources\images along witht he rest of the images.

cerulyeann committed:
Updated the Ui.png image to more accurately reflect the current ui.

cerulyeann committed:
Added the jar file to run the todo list.

cerulyeann committed:
feat: add 'help' command for CirnoBot usage guide

Added a new 'help' command that provides a brief guide on how to use CirnoBot.

cerulyeann committed:
docs(Task): update Javadocs for better clarity

Updated Javadocs in Task.java to improve documentation and readability.

cerulyeann committed:
docs(Task): update Javadocs for better clarity

Updated Javadocs in Task.java to improve documentation and readability.

cerulyeann committed:
docs(Task): update Javadocs for better clarity

Updated Javadocs in Task.java to improve documentation and readability.

cerulyeann committed:
docs(Task): update Javadocs for better clarity

Updated Javadocs in Task.java to improve documentation and readability.

cerulyeann committed:
docs(Task): update Javadocs for better clarity

Updated Javadocs in Task.java to improve documentation and readability.

cerulyeann committed:
docs(HandleFindCommand): update Javadocs for better clarity

Updated Javadocs in HandleFindCommand.java to improve documentation and readability.

cerulyeann committed:
docs(HandleDeleteCommand): update Javadocs for better clarity

Updated Javadocs in HandleDeleteCommand.java to improve documentation and readability.

cerulyeann committed:
docs(HandleDeleteCommand): update Javadocs for better clarity

Updated Javadocs in HandleDeleteCommand.java to improve documentation and readability.

cerulyean committed:
Merge pull request #2 from cerulyean/branch-Merging-PRs

docs(HandleDeleteCommand): update Javadocs for better clarity

cerulyean committed:
Update README.md

Since the help command is implemented, the disclaimer that it isn't has been removed

cerulyeann committed:
chore(dependencies): add Linux and Mac support for JavaFX dependencies

Updated the dependencies to include Linux and Mac classifiers for JavaFX modules alongside the existing Windows support.

cerulyeann committed:
feat: Fix UI resizing and add tiling background image

- Fixed resizing for MainWindow and DialogBox.
- Added tiling behavior for background image.

cerulyeann committed:
feat(tasks): add mark/unmark commands with existing state checks

Implemented mark and unmark commands that recognize when a task is already marked or unmarked.

cerulyeann committed:
feat(tasks): add mark/unmark commands with existing state checks

Implemented mark and unmark commands that recognize when a task is already marked or unmarked.
